src/pandapipes/component_models/heat_consumer_component.py

```diff
@@ -119,50 +137,60 @@ def adaption_after_derivatives_hydraulic(cls, net, branch_pit, node_pit, idx_loo
         # set all pressure derivatives to 0 and velocity to 1; load vector must be 0, as no change
         # of velocity is allowed during the pipeflow iteration
         f, t = idx_lookups[cls.table_name()]
-        fc_branch_pit = branch_pit[f:t, :]
-        fc_array = get_component_array(net, cls.table_name())
-        # TODO: this is more precise, but slower:
-        #       np.isin(fc_array[:, cls.MODE], [cls.MF_QE, cls.MF_DT, cls.MF_TR])
-        mdot_controlled = ~np.isnan(fc_array[:, cls.MASS])
-        fc_branch_pit[mdot_controlled, JAC_DERIV_DP] = 0
-        fc_branch_pit[mdot_controlled, JAC_DERIV_DP1] = 0
-        fc_branch_pit[mdot_controlled, JAC_DERIV_DM] = 1
-        fc_branch_pit[mdot_controlled, LOAD_VEC_BRANCHES] = 0
-
-    # @classmethod
-    # def adaption_before_derivatives_thermal(cls, net, branch_pit, node_pit, idx_lookups, options):
-    #     f, t = idx_lookups[cls.table_name()]
-    #     hs_pit = branch_pit[f:t, :]
-    #     mask_t_return = ~np.isnan(hs_pit[:, TRETURN])
-    #     hs_pit[mask_t_return, TINIT_OUT] = (hs_pit[mask_t_return, TINIT_OUT]
-    #                                         - hs_pit[mask_t_return, DELTAT])
-    #
-    #
-    # @classmethod
-    # def adaption_after_derivatives_thermal(cls, net, branch_pit, node_pit, idx_lookups, options):
-    #     """
-    #
-    #     :param net:
-    #     :type net:
-    #     :param branch_component_pit:
-    #     :type branch_component_pit:
-    #     :param node_pit:
-    #     :type node_pit:
-    #     :return:
-    #     :rtype:
-    #     """
-    #     # -(rho * area * cp * v_init * (-t_init_i + t_init_i1 - tl)
-    #     #   - alpha * (t_amb - t_m) * length + qext)
-    #
-    #     f, t = idx_lookups[cls.table_name()]
-    #     hs_pit = branch_pit[f:t, :]
-    #     from_nodes = hs_pit[:, FROM_NODE_T].astype(np.int32)
-    #
-    #     mask_qext = ~np.isnan(hs_pit[:, QEXT])
-    #     mask_deltat = ~np.isnan(hs_pit[:, DELTAT])
-    #     mask_t_return = ~np.isnan(hs_pit[:, TRETURN])
-    #     mask_mass = ~np.isnan(hs_pit[:, MASS])
-    #     hs_pit[mask_t_return | mask_deltat, JAC_DERIV_DT1] = 0
+        consumer_array = get_component_array(net, cls.table_name())
```

Comment:
We should add a test with some consumers oos and different control strategies... I am wondering if this is working properly, as the hc_pit here is the active pit.

Reply:
Ah, just saw that the default is get_component_array only refers to active pit